### DISCUSSION TITLE
how to create diffs in subsheets

### DISCUSSION THREAD

**Discussion by harm363** [INTI-CMNB/KiBot#333] [Q&A] [answered]
I have projects that contain a number of subsheets in K6, i want to create a schematic diff of every sheet from the current state to HEAD-1. 

the project layout is:
source/ 
  main.kicad_sch
  subsheet1.kicad_sch
  subsheetx.kicad_sch

I made a script that finds all sheets and writes them into a kibot.yaml file from a template.
find_schematics.sh:

    #! /bin/bash
    #finds all the schematics in the sourcemap and creates a kibot.yaml file to generate a diff for all schematics
    PRJROOT=$PWD
    CI=ZUH-CI 
    CIPATH=$PRJROOT/.${CI}
    CONFIGS=${CIPATH}/kibot_config
    SCRIPTS=${CIPATH}/scripts
    SOURCE=$PRJROOT/source

    find_schematic_files() {
        sourcepath=$1
        #write kibot yaml header in new file
        echo -e "kibot:\n version: 1\noutputs:" > $CONFIGS/local.schdiff.config.kibot.yaml
        for sheetname in $sourcepath/*kicad_sch 
        do 
           echo "find sheet $sheetname"
           sheet2=$(basename $sheetname)
           file="$CONFIGS/template_schdiff.yaml"
           while IFS= read -r line; do
               line1=$(sed -e "s!{name}!${sheet2}!g") 
               echo -e "$line1" >> $CONFIGS/local.schdiff.config.kibot.yaml
           done <$file
        done
     }
    find_schematic_files $SOURCE 
template_diff:

    #empty line for correct copy purposes
    - name: "schdiff {name}"
        type: diff
        comment: "generate diff file for {name}"
        dir: documentation
        options: 
            output: '%f-%i%-{name}.%x'
            new: {name} #filename 
            new_type: file
            #old: KIBOT_TAG
            #old_type: file
            pcb: false
            force_checkout: true
        priority: 100

but this just diffs the found subsheet as the new sheet to the main sheet like this: (red is the main sheet, green the subsheet)
[main-diff_sch_-sub_abb.kicad_sch.pdf](https://github.com/INTI-CMNB/KiBot/files/10066534/main-diff_sch_-sub_abb.kicad_sch.pdf)

when i define old with the same name as the subsheet i have in new, kibot diffs the same files against eachother (wich would be the expected behaviour of course). 
template_diff:

    #empty line for correct copy purposes
    - name: "schdiff {name}"
        type: diff
        comment: "generate diff file for {name}"
        dir: documentation
        options: 
            output: '%f-%i%-{name}.%x'
            new: {name} #filename 
            new_type: file
            old: {name}
            old_type: file
            pcb: false
            force_checkout: true
        priority: 100

How do I diff 2 subsheets from different commits against eachother? 
like: 
HEAD: subsheet1,kicad_sch vs HEAD-1: subsheet1_kicad_sch

**Comment by harm363** [comment, marked as answer]
I found what is was looking for, schematic diffs are generated with the kicad-diff.py script. the command for it is build by out_diff.py. 
when i manually build the schematic diff: 

      `kicad-diff.py --all_pages ~/Documents/source_old/main.kicad_sch source/main.kicad_sch'
The correct layout is build. kibot seems to be missing the --all_pages option. I will make a issue for the option to be enabled.

**Comment by set-soft** [comment]
Hi @harm363 !
KiBot v1.4.1 and newer compares sub-sheets, is the default behavior.
This is currently in the `dev` branch.

**Reply by harm363** [reply]
nice, I tested it, it works fine for met
